test/SerialIOTest.cpp

```diff
@@ -1629,8 +1629,24 @@ void fileBased_write_test(const std::string & backend)
         REQUIRE(o.iterations[5].time< double >() == 5.0);
     }
 
-    // extend existing series with new step and auto-detection of iteration padding
+    if( backend == "bp" )
     {
+        // Append + filebased iteration encoding works for all backends
```

Comment:
The `if "bp"` above is confusing (outdated?) if we want to run this for all backends now :)

Reply:
Outdated, yeah
While working on the PR, read_write mode was for a while completely inaccessible for ADIOS2 because I had planned to replace it with Append mode entirely
Then I noticed that there are still reasons to use read_write with ADIOS2 in file-based iteration encoding, so those specialized test paths can now be deleted again